http.js

```diff
@@ -393,11 +392,11 @@ exports.read = function (request, qualifier) {
             if (Buffer.isBuffer(data)) {
                 contentType = response.headers['content-type'] || '',
                 match = contentType.match(/charset=([a-z0-9-]+)/i),
-                charset = match ? match[1] : 'utf8';
+                charset = match && iconv.encodingExists(match[1]) ? match[1] : 'utf8';
```

Comment:
That's the default behavior of iconv-lite, #141 actually does that. 

I've wapped it into encodingExists() to prevent the error - as far as HTTP libraries are concerned, I've never seen one throwing an "encoding unknown" error. They all fallback to some sort of default (UTF-8 being a pretty good one, all things considered).